core/src/com/fourinachamber/fortyfive/FortyFive.kt

```diff
@@ -66,17 +63,20 @@ object FortyFive : Game() {
 
         serviceThread.sendMessage(ServiceThreadMessage.PrepareResources)
 
-        if (currentScreen == null) {
+        fun onScreenChange() {
             FortyFiveLogger.title("changing screen")
+            currentScreen?.dispose()
             this.currentScreen = screen
             currentRenderable = screen
             setScreen(screen)
+            // TODO: not 100% clean, this function is sometimes called when it isn't necessary
+            MapManager.invalidateCachedAssets()
+        }
+
+        if (currentScreen == null) {
+            onScreenChange()
         } else currentScreen.afterMs(currentScreen.transitionAwayTime ?: 0) {
-            FortyFiveLogger.title("changing screen")
-            currentScreen.dispose()
-            this.currentScreen = screen
-            currentRenderable = screen
-            setScreen(screen)
+            onScreenChange()
         }
```

Comment:
maybe git hows it weird, but doesn't it do the same in both ifs?  (at least the "onScreenChange()" can be extracted outside

Reply:
Its not actually the same because of the `curScreen.afterMs` function call